sycl/test/devicelib/math_test.cpp

```diff
@@ -9,64 +9,821 @@ namespace s = cl::sycl;
+      cgh.single_task<class DeviceLog1p>([=]() {
+        float a = 0;
+        res_access[0] = log1pf(a);
+      });
+    });
+  }
+
+  assert(result == ref);
+}
+
+class DeviceLog2;
+
+void device_log2_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  float ref = 0;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceLog2>([=]() {
+        float a = 1;
+        res_access[0] = log2f(a);
+      });
+    });
+  }
+
+  assert(result == ref);
+}
+
+class DeviceLogb;
+
+void device_logb_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  float ref = 0;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceLogb>([=]() {
+        float a = 1;
+        res_access[0] = logbf(a);
+      });
+    });
+  }
+
+  assert(result == ref);
+}
+
+class DeviceRemainder;
+
+void device_remainder_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  float ref = 0.5;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceRemainder>([=]() {
+        float a = 0.5;
+        float b = 1;
+        res_access[0] = remainderf(a, b);
+      });
+    });
+  }
+
+  assert(result == ref);
+}
+
+class DeviceRemquo;
+
+void device_remquo_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  int quo = -1;
+  float ref = 0.5;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    s::buffer<int, 1> buffer2(&quo, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      auto quo_access = buffer2.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceRemquo>([=]() {
+        float a = 0.5;
+        float b = 1;
+        res_access[0] = remquof(a, b, &quo_access[0]);
+      });
+    });
+  }
+
+  assert(result == ref);
+}
+
+class DeviceTgamma;
+
+void device_tgamma_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceTgamma>([=]() {
+        float a = NAN;
+        res_access[0] = tgammaf(a);
       });
     });
   }
 
-  assert(result_f == 0);
+  assert(std::isnan(result));
 }
 
 void device_math_test(s::queue &deviceQueue) {
   device_cos_test(deviceQueue);
```

Comment:
This will be terribly slow.
We need to make LIT tests as fast possible.
Also each test case duplicates a ton of a boilerplate code.
First idea:
- put all math function calls into a single kernel.